### PR TITLE
Add non_kube_system_pod_with_host_mount query for Kubernetes

### LINKED ISSUE #622: Add non_kube_system_pod_with_host_mount query for Kubernetes
### Platform
*Kubernetes*

### Description
*Ensure pods outside of kube-system do not have access to node volume*

### PR DESCRIPTION
**Pull request by rogeriopeixotocx**
Ensure pods outside of kube-system do not have access to node volume (Deployments, DaemonSets and other templated types too)

Closes #622

### CODE REVIEW

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "Non_Kube_System_Pod_With_Host_Mount",
+  "queryName": "Non Kube System Pod With Host Mount",
+  "severity": "MEDIUM",
+  "category": null,
+  "descriptionText": "Ensure pods outside of kube-system do not have access to node volume",
+  "descriptionUrl": "#"
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/metadata.json:7]: Add url "https://kubernetes.io/docs/concepts/policy/pod-security-policy/"

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package Cx
+
+CxPolicy [result] {
+  resource := input.document[i]
+  metadata := resource.metadata
+  not metadata.namespace
+  kind := resource.kind
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego:7]: FYI, if you just use it once, you don't need to define a variable. Just a heads up, don't change if you don't want to.

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  kind == "Pod"
+  volumes := resource.spec.volumes
+  volumes[_]["hostPath"]
+  result := {
+    "documentId": input.document[i].id,
+    "searchKey": sprintf("metadata.name=%s.spec.volumes.name=%s.hostPath", [metadata.name, volumes[_].name]),
+    "issueType": "IncorrectValue",
+    "keyExpectedValue": "Ensure pods outside of kube-system do not have access to node volume",
+    "keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, "default"])
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego:16]: "keyExpectedValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, "default"]),
"keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' has a hostPath mounted", [resource.metadata.name, kind, "default"])

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  kind := resource.kind
+  kind == "Pod"
+  volumes := resource.spec.volumes
+  volumes[_]["hostPath"]
+  result := {
+    "documentId": input.document[i].id,
+    "searchKey": sprintf("metadata.name=%s.spec.volumes.name=%s.hostPath", [metadata.name, volumes[_].name]),
+    "issueType": "IncorrectValue",
+    "keyExpectedValue": "Ensure pods outside of kube-system do not have access to node volume",
+    "keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, namespace])
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego:34]: "keyExpectedValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, namespace]),
"keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' has a hostPath mounted", [resource.metadata.name, kind, namespace])

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  kind != "Pod"
+  spec := resource.spec.template.spec
+  volumes := spec.volumes
+  volumes[_]["hostPath"]
+  result := {
+    "documentId": input.document[i].id,
+    "searchKey": sprintf("metadata.name=%s.spec.template.spec.volumes.name=%s.hostPath", [metadata.name, volumes[_].name]),
+    "issueType": "IncorrectValue",
+    "keyExpectedValue": "Ensure pods outside of kube-system do not have access to node volume",
+    "keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace actual '%s' should not have hostPath mounted", [resource.metadata.name, kind, namespace])
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego:54]: "keyExpectedValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, namespace]),
"keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' has a hostPath mounted", [resource.metadata.name, kind, namespace])

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  kind != "Pod"
+  spec := resource.spec.template.spec
+  volumes := spec.volumes
+  volumes[_]["hostPath"]
+  result := {
+    "documentId": input.document[i].id,
+    "searchKey": sprintf("metadata.name=%s.spec.template.spec.volumes.name=%s.hostPath", [metadata.name, volumes[_].name]),
+    "issueType": "IncorrectValue",
+    "keyExpectedValue": "Ensure pods outside of kube-system do not have access to node volume",
+    "keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace actual '%s' should not have hostPath mounted", [resource.metadata.name, kind, "default"])
```
Comment (Ruben-Silva) [inline assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego:72]: "keyExpectedValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' should not have hostPath mounted", [resource.metadata.name, kind, "default"]),
"keyActualValue": sprintf("Resource name '%s' of kind '%s' in a non kube-system namespace '%s' has a hostPath mounted", [resource.metadata.name, kind, "default"])

**Review by Ruben-Silva — CHANGES_REQUESTED** [review]

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego (hunk 4 above)
Comment (rogeriopeixotocx) [reply assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego]: solved

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego (hunk 2 above)
Comment (rogeriopeixotocx) [reply assets/queries/k8s/non_kube_system_pod_with_host_mount/query.rego]: ok!

File: assets/queries/k8s/non_kube_system_pod_with_host_mount/metadata.json (hunk 1 above)
Comment (rogeriopeixotocx) [reply assets/queries/k8s/non_kube_system_pod_with_host_mount/metadata.json]: done!

**Comment by rogeriopeixotocx** [comment]
@Ruben-Silva requested changes addressed

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**